lib/tokens.ts

```diff
@@ -25,50 +25,66 @@ export async function enrichTokenMetadata(
   rpcProvider: Provider
 ): Promise<TokenProps & { isFallback?: boolean }> {
   // get the mint
-  const mint = await getMint(rpcProvider.connection, tokenAddress);
+  try {
+    const mint = await getMint(rpcProvider.connection, tokenAddress);
+
+    // second check jup list
+    const tokenOnJup = await getTokenFromJupStrictList(tokenAddress);
+    if (tokenOnJup) {
+      return {
+        symbol: tokenOnJup.symbol,
+        publicKey: tokenAddress.toString(),
+        url: tokenOnJup.logoURI,
+        decimals: tokenOnJup.decimals,
+        name: tokenOnJup.name,
+      };
+    }
+
+    // next, try metaplex
+    const jsonMetadata = await getMetaplexMetadataForToken(
+      tokenAddress,
+      rpcProvider
+    );
+    if (jsonMetadata && jsonMetadata.symbol) {
+      return {
+        symbol: jsonMetadata.symbol,
+        publicKey: tokenAddress.toString(),
+        url: jsonMetadata.image,
+        decimals: jsonMetadata.seller_fee_basis_points,
+        name: jsonMetadata.name,
+      };
+    }
+
+    // next, try token keg and token 2022
```

Comment:
Just FYI this isn't trying tokenkeg, this is just token2022 the getmetadata is ONLY called for token2022 otherwise you want tokenkeg to use getMint()

Reply:
This is the only change requested.

Reply:
are we sure we want to use this @solana/spl-token-registry package? The package seems to essentially a giant json file anyway, so the size of it is 4.4MB minified.

Reply:
No